sklearn/manifold/t_sne.py

```diff
@@ -518,15 +518,27 @@ class TSNE(BaseEstimator):
         learning rate is too low, most points may look compressed in a dense
         cloud with few outliers. If the cost function gets stuck in a bad local
         minimum increasing the learning rate may help.
+        Some discussion on how to set learning rate optimally can be found
+        at https://doi.org/10.1101/451690. Effective use of this parameter has
+        the benefit of decreasing the number of iterations required
+        for a good embedding.
 
     n_iter : int, optional (default: 1000)
-        Maximum number of iterations for the optimization. Should be at
-        least 250.
+        Maximum number of iterations for the optimization. Must be greater than
+        or equal to n_iter_early_exag. If embedding quality is suffering as a
+        consequence of increasing number of samples being embedded, increasing
+        this value and n_iter_early_exag proportionately can help.
+
+    n_iter_early_exag : int, optional (default: 250)
+        Number of iterations out of total n_iter that t-SNE should spend
+        in the early exaggeration phase. If embedding quality is suffering as a
+        consequence of increasing number of samples being embedded, increasing
+        this value and n_iter proportionately can help.
```

Comment:
would it often be useful to specify this as a fraction of n_iter?

Reply:
That is one way to do it but I think it is better to specify the number as the actual number. This is for two main reasons:

1. Aligns with other existing implementations which have the equivalent of this argument specified as the number (see my note about LvdM implementation above).
2. Ensures that existing client code will get the same results as before, regardless of the number of iterations they had set with the n_iter argument. If we suddenly start using a percentage as the default argument, code with n_iter set to something different than 1000 will suddenly use a different early exaggeration phase length.